apps/spruce/src/pages/waterfall/caching/index.ts

```diff
@@ -0,0 +1,145 @@
+import { FieldMergeFunction, FieldReadFunction } from "@apollo/client";
+import { WaterfallQuery } from "gql/generated/types";
+import { VERSION_LIMIT } from "../constants";
+
+export const readVersions = ((existing, { args, readField }) => {
```

Comment:
i might be doing something wrong but i think the cache might not be working! i keep getting the loading screen between pagination. maybe the new field also needs to be returned in the read function?

Reply:
duh, added

Reply:
I tightened up the typing of the caching functions so that future similar changes would not slip through the cracks 👍